micro-frontends/src/next-ui/utils/PatientAllergiesControl/AllergyControlUtils.js

```diff
@@ -0,0 +1,66 @@
+import axios from "axios";
+import { BAHMNI_ENCOUNTER_URL, ENCOUNTER_TYPE_URL, FETCH_CONCEPT_URL, GET_ALLERGIES_URL, SAVE_ALLERGIES_URL } from "../../constants";
+import { getLocale } from "../../Components/i18n/utils";
+
+export const fetchAllergensOrReactions = async (conceptId) => {
+  const locale = getLocale();
+  const apiURL = FETCH_CONCEPT_URL.replace("{locale}", locale);
+
+  const replaceConceptIdInapiURL = (conceptId) => {
+    return apiURL.replace("{conceptUuid}", conceptId);
+  };
+
+  const allergenOrReactionURL = replaceConceptIdInapiURL(conceptId);
+
+  try {
+    const response = await axios.get(allergenOrReactionURL);
+    return response.data;
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+export const fetchAllergiesAndReactionsForPatient = async (patientId) => {
+  try {
+    const response = await axios.get(GET_ALLERGIES_URL.replace("{patientId}", patientId));
+    return response.data;
+  }
+  catch (error) {
+    console.log(error);
```

Comment:
Priority: MAJOR

Using an error visualizer such as a modal is preferred because users, including testers in a staging environment, typically do not have access to the console. Displaying console logs in production could expose sensitive information about the internal workings of the system.

In a production environment, if an error is due to a user's action or may impact the user, it's important to display a message informing them. Therefore, we recommend avoiding console logging in both staging and production environments to maintain security and consistency.